int8 recv_flag
